src/hotspot/share/services/memReporter.hpp

```diff
@@ -83,10 +83,17 @@ class MemReporterBase : public StackObj {
 
     size_t amount = s1 - s2;
     assert(amount <= SIZE_MAX - _scale / 2, "size_t overflow");
-    amount = (amount + _scale / 2) / _scale;
-    // We assume the valid range for deltas [INT64_MIN, INT64_MAX] to simplify the code.
-    assert((sizeof(size_t) < sizeof(int64_t)) || (sizeof(size_t) == sizeof(int64_t) && amount <= INT64_MAX), "cannot fit scaled diff into size_t");
-    return (is_negative) ? -(int64_t)amount : (int64_t)amount;
+    size_t scaled = (amount / _scale);
+    if ((amount % _scale) > (_scale - 1)/2) {
+      scaled += 1;
```

Comment:
We need an assert `assert(scaled < SIZE_MAX, "size_t overflow");` for the case `_scale == 1`.

Reply:
We do not need such an assert.  If `_scale == 1` then `amount % _scale` is 0 and the condition is false,
so no increment will occur.

Reply:
You are right no assert is needed.